game-core/src/main/java/games/strategy/engine/random/PbemDiceRoller.java

```diff
@@ -258,5 +258,13 @@ private void rollInSeperateThread() throws IllegalStateException {
         notifyError();
       }
     }
+
+    private void waitForWindowToBecomeVisible() {
+      final BooleanSupplier isVisible =
+          () -> Interruptibles.awaitResult(() -> SwingAction.invokeAndWaitResult(this::isVisible)).result.orElse(false);
```

Comment:
To be technically correct, the `Component#isVisible()` method should be called on the EDT.  The `Component#visible` field is not always accessed in a thread-safe manner within the `Component` class.